libs/sql-parser/src/test/java/io/crate/sql/IdentifiersTest.java

```diff
@@ -64,6 +64,7 @@ public void testQuoteIfNeeded() {
         assertThat(Identifiers.quoteIfNeeded("fhjgadhjgfhs")).isEqualTo("fhjgadhjgfhs");
         assertThat(Identifiers.quoteIfNeeded("fhjgadhjgfhsÖ")).isEqualTo("\"fhjgadhjgfhsÖ\"");
         assertThat(Identifiers.quoteIfNeeded("ABC")).isEqualTo("\"ABC\"");
+        assertThat(Identifiers.quoteIfNeeded("\"ABC\"")).isEqualTo("\"ABC\""); // quoted-ident containing upper case letters
```

Comment:
I think this is not correct, we need to remove the enclosing `"` from the identifier name. 
Postgres:
```
matriv=> create schema TeSt;
CREATE SCHEMA

matriv=> create table test.my_table(a int);
CREATE TABLE

matriv=> SELECT schema_name
FROM information_schema.schemata;
    schema_name     
--------------------
 pg_catalog
 public
 information_schema
 test
(4 rows)

matriv=> select * from test.my_table;
 a 
---
(0 rows)


matriv=> create schema "TeST";
CREATE SCHEMA

matriv=> SELECT schema_name 
FROM information_schema.schemata;
    schema_name     
--------------------
 pg_catalog
 public
 information_schema
 test
 TeST
(5 rows)

matriv=> create table "TeST".my_table(a int);
CREATE TABLE
matriv=> select * from "TeST".my_table;
 a 
---
(0 rows)

```

Reply:
Thank you @matriv could this be handled by [d1caca7](https://github.com/crate/crate/pull/13454/commits/d1caca7f5ccbc10537bf77d7324711180b5a4ca5) ?

Reply:
This only removes the `""` when displaying the schema name, I believe it's not correct to keep them there, to begin with, they are not part of the identifier, the are helper markers to show that what's enclosed must be treated differently. @seut do you agree?

Reply:
Yes I agree, they should not be stored anyway. They should just be marker to define if the identifiers are converted to lowercase/treat-as-case-sensitive or not.

Reply:
Maybe the confusion comes from: How should case-sensitive identifier be differentiated later on from non case-sensitive ones? -> I don't think that this is needed, when showing/outputting we can always quote them.